packages/tasks/src/tasks/text-generation/inference.ts

```diff
@@ -3,7 +3,6 @@
  *
```

Comment:
(commenting on `inference.ts` but more related to `./spec`)

TGI server has an OpenAPI documentation to document its inputs and outputs (see [swagger ui](https://huggingface.github.io/text-generation-inference/#/Text%20Generation%20Inference/generate) and [openapi.json](https://huggingface.github.io/text-generation-inference/openapi.json)). I've spotted a few differences compared to the specs here. Some of these properties are currently specific to TGI and not available in the "transformers-based inference" endpoints.

In `TextGenerationParameters`:
- `max_new_tokens` (integer) => Maximum number of generated tokens.
- `repetition_penalty` (number) => The parameter for repetition penalty. A value of 1.0 means no penalty. See [this paper](https://arxiv.org/pdf/1909.05858.pdf) for more details.
- `seed` (integer) => Random sampling seed.
- `stop` (array of string) => Stop generating tokens if a member of `stop` is generated. (currently named `stop_sequences` in these specs / `stop` in TGI => same purpose => to harmonize?)
- `top_n_tokens` (integer) => ??? (no idea what this is. We don't have it either in Python client)

In the Python client, along with the `inputs` (string) + `parameters`  (TextGenerationParameters) values in the request payload we also have a `stream` (bool) value. If True the output type is different (same as `/generate_stream` instead of `/generate` in TGI). I don't see it documented [here](https://huggingface.github.io/text-generation-inference) so I wonder if these docs might be a bit outdated (ping @Narsil you might know more?). In any case, this is an option currently working and used so worth keeping it.

In `TextGenerationOutputDetails`:
- `best_of_sequences` (array of BestOfSequence) => Additional sequences when using the `best_of` parameter.

`BestOfSequence` (currently missing):
- `generated_text` (string) => The generated text.
- `finish_reason` (FinishReason) => The reason for the generation to finish, represented by a FinishReason value.
- `generated_tokens` (integer) => The number of generated tokens in the sequence.
- `seed` (integer) => The sampling seed if sampling was activated.
- `prefill` (array of InputToken) => he decoder input tokens. Empty if `decoder_input_details` is False.
- `tokens` (array of Token) => The generated tokens.

`BestOfSequence` object is actually very similar to `TextGenerationOutputDetails` but 1 level of nesting below.

Reply:
I deliberately left out the `stream` parameter because it was API/transport-specific (and not inference-specific)

I'm happy to revisit that if there's a consensus tho